File: Server/hms/src/main/java/com/hms/prescription/model/Prescription.java
package com.hms.prescription.model;

import com.hms.common.dto.Tables;
import com.hms.common.model.Auditable;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.util.Date;
import java.util.List;

@Entity
@Table(name = Tables.PRESCRIPTION)
@Getter
@Setter
public class Prescription extends Auditable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    private Integer doctorId;

    private Integer patientId;

    private Date prescriptionDate;

    private String status;

    private String bloodPressure;

    private String bloodSugar;

    @OneToMany(cascade = CascadeType.ALL,fetch = FetchType.EAGER,orphanRemoval = true)
    @JoinColumn(name = "prescription_id")
    private List<Medicine> medicines;


}

File: Client/hms/src/app/models/Prescription.ts
import { Address } from "cluster";
import { Inventory } from "./Inventory";
import { User } from "./user.model";

export class Prescription{
    id:number | undefined;
    patientId: number |undefined;
    doctorId:number |undefined;
    doctorName:string = '';
    patientName:string ='';
    note:string='';
    status:string='';
    bloodPressure:string='';
    bloodSugar:string='';
    medicines: Medicine[] = [];
    prescriptionDate:number | undefined;
    doctor:User = new User();
    patient:User = new User() ;
}

export class Medicine{
    id: number | undefined;
    inventoryId: number | undefined;
    inventory : Inventory | undefined;
    dosage:string = '';
    frequency:string ='';
    duration:string ='';
    note:string='';
    
}